### PR TITLE
feat: update contracts to v1.0.0

### PR DESCRIPTION
**Pull request by Meyanis95**
## Motivation

This PR updates the AnonAadhaar contracts.
It ensures that the `publicKey` and the `signal` are right and then verifies the groth16 proof by calling the `Verifier.sol` contract.
It also adds an Example contract AnonAadhaarVote to show how the Soldity verifier should be used by devs. 

## Change Summary

- Fixed scripts to generate groth16 Solidity verifier.
- `AnonAadhaar.sol` verifier make sure that pubKey and signal are right and call Verifier.
- `AnonAadhaarVote.sol`, an example use case of the solidity verifier.
- Vote tests are under the same file as AnonAadhaar tests to gain time for the CI/CD by generating only one proof.
- Contracts deployed on Sepolia.
- Updates core packages from Vu and Saleel comments: 
   - Add signalHash to claim.
   - Hide PCD from core package.

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [change set](https://github.com/privacy-scaling-explorations/anon-aadhaar/CHANGELOG.md)
- [x] PR has been tagged with a change label(s) (i.e. documentation, feature, bug fix, or chore)
- [x] PR includes documentation if necessary.

## Additional Context

Contracts deployed on Sepolia for both production and test public keys:
```
{
  "Verifier": "0x3Bbb6EC5F9Fd4912AC0E181887ac0545B3C542c1",
  "testPublicKey": {
    "AnonAadhaarTest": "0x9394E7661bd0031F00D580391f2A3089C63F4680",
    "AnonAadhaarVote": "0x1b51d2067205ef73e3D122fE9a3dfdcBDe8300CC"
  },
  "productionPublicKey": {
    "AnonAadhaarTest": "0x42faf48F11644f0C7942b21aD707a8058E2ad1bb",
    "AnonAadhaarVote": "0xde1d63eB664D18Dcc76fADDb1cB54Ec01752E04d"
  }
}
```

Gas report from test:
image

### CODE REVIEW

File: packages/contracts/interfaces/IAnonAadhaarVote.sol
Comment (saleel) [inline packages/contracts/interfaces/IAnonAadhaarVote.sol:17]: Is there an advantage on having an inerface to derive functions? We can keep the structs and event only and import them in the implementation contract (instead of that contract deriving this interface). I think interface like this is relevant when we expect to have other implementation of the all signature. wdyt?

File: packages/contracts/contracts/example/AnonAadhaarVote.sol
Comment (Meyanis95) [reply packages/contracts/contracts/example/AnonAadhaarVote.sol:1]: You can only give one directory to the hardhat compiler, so in that case `AnonAadhaarVote` couldn't be compiled.
What about `src/example`?

File: packages/contracts/contracts/AnonAadhaar.sol
Comment (Meyanis95) [reply packages/contracts/contracts/AnonAadhaar.sol:27]: I agree that the signal is just a commitment and it can be used any hash function. But as we want to ensure that the resulting hash is constrained by the snark field, I thought it's better to provide the hashing method as part of the SDK/Protocol, making the life of devs easier. 

Also while you can just send the signal hashed to the verifier you might want to check the data that has been committed while generating the proof, to prevent it from front-running, that's why I added the hash method so any contract above can simply send to the verifier the plain signal, after having it checked it, and the verifier contract will then compute the hash of the committed value to verify the gorth16 proof.
This way the check of the plain signal value is delegated to the contract logic and not to the verifier, making it easier for devs to use `signal` as they want for their use case.

File: packages/contracts/contracts/AnonAadhaar.sol
Comment (Meyanis95) [reply packages/contracts/contracts/AnonAadhaar.sol:43]: Great idea!
I didn't think about it.
I'll change that

File: packages/contracts/deployed-contracts/goerli.json
Comment (Meyanis95) [reply packages/contracts/deployed-contracts/goerli.json:1]: Will do!

File: packages/contracts/contracts/AnonAadhaar.sol
Comment (saleel) [reply packages/contracts/contracts/AnonAadhaar.sol]: Ok, got it 👍

File: packages/core/src/types.ts
Comment (vuvoth) [inline packages/core/src/types.ts:58]: WE should change WitnessQRInputs have same properties with AnonAadhaarArgs

File: packages/core/test/core.test.ts
Comment (vuvoth) [inline packages/core/test/core.test.ts:21]: Should we set path to aadhaar-verifier is constant, and export it in core circuit packages. 
This help developer can import it more easier.

File: packages/react/test/hook.test.tsx
Comment (vuvoth) [inline packages/react/test/hook.test.tsx:29]: I think we must set 512 * 3 is constant, maybe MAX_QR_LEN.

File: packages/contracts/src/example/AnonAadhaarVote.sol
Comment (vuvoth) [inline packages/contracts/src/example/AnonAadhaarVote.sol:44]: Should we check timestamp here?

File: packages/contracts/test/Verifier.ts
Comment (vuvoth) [inline packages/contracts/test/Verifier.ts:30]: I think we should export gov certification.  Because this public data and developer also use it for check something(like check signature before gen the proof). 
I remember we have 2 or 3 certs.

File: packages/core/src/types.ts
Comment (Meyanis95) [reply packages/core/src/types.ts:58]: I will remove it, as it is no longer used.

File: packages/contracts/test/Verifier.ts
Comment (Meyanis95) [reply packages/contracts/test/Verifier.ts:30]: I didn't fully understand, do you mean allowing importing certificates from the SDK?

Comment (vuvoth) [reply packages/contracts/test/Verifier.ts:30]: Yes, so developer don't need find it by themself.

Comment (Meyanis95) [reply packages/contracts/test/Verifier.ts:30]: I'm not sure it's the proper way to handle it there. Also, it might not be in the scope of this PR.

Comment (vuvoth) [reply packages/contracts/test/Verifier.ts:30]: Got it! I think this useful because developer need cert to gen the proof. 
Lets do it in another PR!

File: packages/contracts/src/example/AnonAadhaarVote.sol
Comment (Meyanis95) [reply packages/contracts/src/example/AnonAadhaarVote.sol:44]: That's a good point!
I wonder if it's something that should be implemented here, or just should be specified in the documentation, as it will be hard for us to have a proper test environment to test it.

Comment (vuvoth) [reply packages/contracts/src/example/AnonAadhaarVote.sol:44]: We can do it later. 
I remember in hardhat we can set timestamp so maybe this help we test smart contract easier.

Comment (saleel) [reply packages/contracts/src/example/AnonAadhaarVote.sol:44]: Good points. If its difficult for tests we can simply add a comment and commented out code to handle timestamp. Since this is an example others referring to it can implement properly.

Comment (Meyanis95) [reply packages/contracts/src/example/AnonAadhaarVote.sol:52]: I added it without making the test fail as suggested by @vuvoth. Just changed the starting block of the EVM fork running while testing.

File: packages/contracts/hardhat.config.ts
Comment (vuvoth) [inline packages/contracts/hardhat.config.ts:10]: I don't think we should use main net here

Comment (Meyanis95) [reply packages/contracts/hardhat.config.ts:10]: Can you provide more information? And what should be instead of `mainnet`?

This is a specification for `hardhat` networks, this is used to run tests from a `mainnet` fork that starts from the block `7327025`, to be able to emulate the conditions for the proof generated, with the test QR code with the timestamp `20190308114407437`, to be validated. This does not impact the deployment flow, and it's only a setting for hardhat while running the test. 
Also, Sepolia could not be set here, as it does not have blocks minted before 2021.

Comment (vuvoth) [reply packages/contracts/hardhat.config.ts:10]: Oh, Got it! Thank you explanation!

**Comment by vuvoth** [comment]
LGTM! Waiting for @saleel approval!

File: packages/contracts/src/example/AnonAadhaarVote.sol
Comment (Meyanis95) [reply packages/contracts/src/example/AnonAadhaarVote.sol:50]: As it implies also changes in the whole flow of how to compute the `signalHash`, I created another ticket for this #154

**Review by saleel — APPROVED** [review]

**Pull request merged**
